File: src/main/java/pl/training/module6/observer/after/ReleasePublished.java
package pl.training.module6.observer.after;

public record ReleasePublished(String releaseId) {
    public ReleasePublished {
        if (releaseId == null || releaseId.isBlank()) {
            throw new IllegalArgumentException("releaseId must not be blank");
        }
    }
}

File: csharp/src/Training.Module6/Observer/After/ReleasePublished.cs
namespace Training.Module6.Observer.After;

public sealed record ReleasePublished
{
    public ReleasePublished(string? releaseId)
    {
        if (string.IsNullOrWhiteSpace(releaseId))
        {
            throw new ArgumentException("releaseId must not be blank");
        }
        ReleaseId = releaseId;
    }

    public string ReleaseId { get; }
}
